app/src/main/java/org/simple/clinic/home/patients/PatientScreenRepository.kt

```diff
@@ -1,17 +1,24 @@
 package org.simple.clinic.home.patients
 
 import io.reactivex.Observable
+import io.reactivex.rxkotlin.ofType
 import org.simple.clinic.home.patients.illustration.HomescreenIllustration
 import org.simple.clinic.storage.files.FileStorage
+import org.simple.clinic.storage.files.GetFileResult
 import javax.inject.Inject
 
 class PatientScreenRepository @Inject constructor(
     private val illustrationDao: HomescreenIllustration.RoomDao,
     private val fileStorage: FileStorage
 ) {
 
-  fun illustrations(): Observable<HomescreenIllustration> =
+  fun illustrations(): Observable<GetFileResult.Success> =
```

Comment:
The repository is now leaking the abstraction that it uses the `FileStorage` component to manage and store images. This will cause an indirect coupling between the `PatientsScreen` and the `FileStorage` class. 

I suggest we can improve this by having the method return an `Observable<Optional<File>>`  instead and map the result from `FileStorage` internally.

Reply:
I realised something about the `AndroidFileStorage` component and the `getFile()` method. It was written with the intention of receiving a `File` handle that would create the file if it didn't exist. However, that is not what we want here because we do not want to show the illustration if the file does not exist. This will end up showing a blank home screen instead of the default image if the image for an occasion has not been downloaded.

This is ok for now since there aren't any occasions being fetched. But I suggest we do two things before we integrate the final API:
- Rename `FileStorage#getFile` to `getOrCreateFile` so the method is explicit about what it does
- Add a new method `FileStorage#file(path)` that returns an `Optional<File>` that can be used for cases like this.